BLSQ/iaso
# IA-3540 restrict payments list by user location

geo-limited users trying to see change request related to a payment could land on an empty page if they didn't have the permission to see the org unit of the change request

Related JIRA tickets : IA-3540

## Self proofreading checklist


## Doc

NA

Explain the changes that were made.

- Add `can_see_change_requests` field to `PotentialPyament` and `PaymentLot` API responses
- In the same APIs add a `can_see_change_request` for each change request
- `can_see_change_request(s)` are computed by comparing the change request org unit with the user's location
- Use these booleans in the front end to remove the link to change request(s):
   - For potential payments
   - For payment lots
   - For individual payments in payment lots 

## How to test

Explain how to test your PR.

DB setup:
- Have approved change requests, with a non-null `created_by` field (may require change to django admin in local)
- Have payment lots with change requests from several org units
- Have a geo-limited user with a location that is not a parent of at least one org unit of the approved change requests and of the payment's change requests, and the payments read-write perm
- Have a super user
- Have a geo-limited user with a location that includes all the org units in all change requests (directly or as parent), and the payments read-write perm

Test:
- Go to Potential payments page:
    -  super user and user with authorized org units should see the action icon button. It should lead them to the change requests' page, where all change requests should be visible
    - geo-limited user should not see the icon button
- Go to payment lots:
    -  super user and user with authorized org units should see the "eye" icon button. Same expected behaviour as potential payments
    - Same expected behaviour as potential payments
- Open/edit a payment lot:
    - super user and authorized user should see the action button ("eye") for all payments
    - limited user should only see it for payments for which all change requests's org units are covered by his location

## Print screen / video

Upload here print screens or videos showing the changes.

## Notes

Things that the reviewers should know:

- known bugs that are out of the scope of the PR
- other trade-offs that were made
- does the PR depends on a PR in [bluesquare-components](https://github.com/BLSQ/bluesquare-components)?
- should the PR be merged into another PR?

## Follow the Conventional Commits specification

The **merge message** of a pull request must follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.

This convention helps to automatically generate release notes.

Use lowercase for consistency.

[Example](https://github.com/BLSQ/iaso/commit/8b8d7d3064138c1e57878f17b4eb922516ab0112):

```
fix: empty instance pop up

Refs: IA-3665
```

Note that the Jira reference is preceded by a _line break_.

Both the line break and the Jira reference are entered in the _Add an optional extended description…_ field.
